audio-format-service/src/main.py

```diff
@@ -0,0 +1,361 @@
+"""Audio Format Service
+
+FastAPI-based service for audio format validation, detection, and conversion.
+Provides REST endpoints for audio format handling operations.
+"""
+
+import os
+import tempfile
+from contextlib import asynccontextmanager
+from typing import Dict, List, Optional
+
+from fastapi import Depends, FastAPI, File, Form, HTTPException, UploadFile, status
+from pydantic import BaseModel, Field
+
+# Import from shared package
+from stt_shared.exceptions import AudioFormatError, ValidationError
+from stt_shared.interfaces.audio_format import AudioFormatHandler
+from stt_shared.mock_audio_format_handler import MockAudioFormatHandler
+
+
+# Response models
+class FormatValidationResponse(BaseModel):
+    """Response model for format validation."""
+
+    valid: bool = Field(..., description="Whether the format is valid")
+    format: Optional[str] = Field(None, description="Detected format")
+    message: Optional[str] = Field(None, description="Validation message")
+
+
+class FormatDetectionResponse(BaseModel):
+    """Response model for format detection."""
+
+    format: str = Field(..., description="Detected audio format")
+    confidence: float = Field(..., description="Detection confidence (0.0-1.0)")
+
+
+class AudioMetadataResponse(BaseModel):
+    """Response model for audio metadata."""
+
+    duration: float = Field(..., description="Duration in seconds")
+    sample_rate: int = Field(..., description="Sample rate in Hz")
+    channels: int = Field(..., description="Number of audio channels")
+    bitrate: Optional[int] = Field(None, description="Bitrate in kbps")
+    format: str = Field(..., description="Audio format")
+    size: int = Field(..., description="File size in bytes")
+
+
+class ConversionResponse(BaseModel):
+    """Response model for format conversion."""
+
+    success: bool = Field(..., description="Whether conversion was successful")
+    original_format: str = Field(..., description="Original audio format")
+    target_format: str = Field(..., description="Target audio format")
+    file_path: Optional[str] = Field(None, description="Path to converted file")
+    message: Optional[str] = Field(None, description="Conversion message")
+
+
+class HealthResponse(BaseModel):
+    """Response model for health check."""
+
+    status: str = Field(..., description="Service status")
+    version: str = Field(..., description="Service version")
+    dependencies: Dict[str, str] = Field(..., description="Dependency status")
+
+
+# Global handler instance
+audio_handler: Optional[AudioFormatHandler] = None
+
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    """Application lifespan manager."""
+    global audio_handler
+
+    # Initialize audio format handler
+    # Use mock handler for now - will be replaced with real implementation
+    audio_handler = MockAudioFormatHandler()
+
+    yield
+
+    # Cleanup
+    audio_handler = None
```

Comment:
move this lifespan app state at least into separate file.
Also based on this small snipped we use lifespan here to initialize dependency injections by given specific values should we follow the best practices of DI initialization, and move all this part, including `get_audio_handler` function into separate file.

I could propose file name "state.py" but not sure whether it is align with best practices of DI and FastAPI recommendations, maybe we have better option here